lib/libutee/tee_api_objects.c

```diff
@@ -397,20 +397,43 @@ TEE_Result TEE_OpenPersistentObject(uint32_t storageID, void *objectID,
 				    uint32_t objectIDLen, uint32_t flags,
 				    TEE_ObjectHandle *object)
 {
-	if (storageID != TEE_STORAGE_PRIVATE)
-		return TEE_ERROR_ITEM_NOT_FOUND;
+	TEE_Result res;
 
-	if (objectID == NULL)
-		return TEE_ERROR_ITEM_NOT_FOUND;
+	if (storageID != TEE_STORAGE_PRIVATE) {
+		res = TEE_ERROR_ITEM_NOT_FOUND;
+		goto err;
+	}
 
-	if (objectIDLen > TEE_OBJECT_ID_MAX_LEN)
-		TEE_Panic(0);
+	if (!objectID) {
+		res = TEE_ERROR_ITEM_NOT_FOUND;
+		goto err;
+	}
 
-	if (object == NULL)
-		return TEE_ERROR_BAD_PARAMETERS;
+	if (objectIDLen > TEE_OBJECT_ID_MAX_LEN) {
+		res = TEE_ERROR_BAD_PARAMETERS;
+		goto err;
+	}
 
-	return utee_storage_obj_open(storageID, objectID, objectIDLen, flags,
+	if (!object) {
+		res = TEE_ERROR_BAD_PARAMETERS;
+		goto err;
+	}
+
+	res = utee_storage_obj_open(storageID, objectID, objectIDLen, flags,
 				     object);
+
+	if (res == TEE_SUCCESS)
+		goto out;
+err:
+	if (res == TEE_ERROR_ITEM_NOT_FOUND ||
```

Comment:
If you added `res == TEE_SUCCESS` here you could get rid of the `if` just above.